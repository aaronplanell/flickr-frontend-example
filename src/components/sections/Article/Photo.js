import React from 'react';

const styleOfPhoto = {
  paddingRight: '5px',
  opacity: 0.9
}

const Photo = ({photo, selectPhoto}) => (
  <img
    alt={photo.id}
    id={photo.id}
    src={photo.source}
    style={styleOfPhoto}
    onClick={(event) => selectPhoto(photo)}
    onMouseOver={(event) => event.target.style.opacity = 1.0}
    onMouseLeave={(event) => event.target.style.opacity = 0.9}
  />
);

export default Photo;
