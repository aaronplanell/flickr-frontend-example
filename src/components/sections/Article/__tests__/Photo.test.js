import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Photo from '../Photo';

const photo = {
  id: 1234,
  original: 'http://host:port/image.jpg'
}

const selectPhoto = jest.fn();

it('Photo Renders OK', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Photo
      photo={photo}
      selectPhoto={selectPhoto}
    />,
    div);
});

it('Photo Snapshot matchs', () => {
  const tree = renderer.create(
    <Photo
      photo={photo}
      selectPhoto={selectPhoto}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
