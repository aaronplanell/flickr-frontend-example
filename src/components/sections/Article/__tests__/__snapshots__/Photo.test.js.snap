exports[`test Photo Snapshot matchs 1`] = `
<img
  alt={1234}
  id={1234}
  onClick={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  src={undefined}
  style={
    Object {
      "opacity": 0.9,
      "paddingRight": "5px",
    }
  } />
`;
